src/aggregator/aggregator/aggregator.go

```diff
@@ -313,19 +316,38 @@ func (agg *aggregator) processPlacementWithLock(
 	if !agg.shouldProcessPlacementWithLock(newStagedPlacement, newPlacement) {
 		return nil
 	}
-	var newShardSet shard.Shards
+	var (
+		notFoundInPlacement = false
+		newShardSet         shard.Shards
+	)
 	instance, err := agg.placementManager.InstanceFrom(newPlacement)
 	if err == nil {
 		newShardSet = instance.Shards()
 	} else if err == ErrInstanceNotFoundInPlacement {
+		notFoundInPlacement = true
 		newShardSet = shard.NewShards(nil)
 	} else {
 		return err
 	}
+
 	agg.updateShardsWithLock(newStagedPlacement, newPlacement, newShardSet)
 	if err := agg.updateShardSetIDWithLock(instance); err != nil {
 		return err
 	}
+
+	if notFoundInPlacement {
+		// NB(r): Without this log message it's hard for operators to debug
+		// logs about receiving metrics that the aggregator does not own.
+		placementInstances := newPlacement.Instances()
+		placementInstanceIDs := make([]string, 0, len(placementInstances))
+		for _, instance := range placementInstances {
+			placementInstanceIDs = append(placementInstanceIDs, instance.ID())
+		}
+		agg.logger.Error("could not find aggregator instance ID in placement",
```

Comment:
nit: maybe also to explicitly call out that instanceId in placement has to match actual instance/pod hostname?

Reply:
Sounds good I'll say "aggregator instance ID must be in placement: missing from placement".  I can't really say pod hostname here since some operators are not on kubernetes or using an environment variable with an arbitrary string for the instance ID.